File: specification/compute/resource-manager/Microsoft.Compute/Compute/stable/2025-11-01/examples-java/virtualMachineExamples/VirtualMachine_Restart_MaximumSet_Gen.java

/**
 * Samples for VirtualMachines Restart.
 */
public final class Main {
    /*
     * x-ms-original-file: 2025-11-01/virtualMachineExamples/VirtualMachine_Restart_MaximumSet_Gen.json
     */
    /**
     * Sample code: VirtualMachine_Restart_MaximumSet_Gen.
     * 
     * @param manager Entry point to ComputeManager.
     */
    public static void virtualMachineRestartMaximumSetGen(com.azure.resourcemanager.compute.ComputeManager manager) {
        manager.serviceClient().getVirtualMachines().restart("rgcompute", "aaaaaaaaaaaaaaaaaaaa",
            com.azure.core.util.Context.NONE);
    }
}

File: specification/compute/resource-manager/Microsoft.Compute/Compute/stable/2025-11-01/examples-js/virtualMachineExamples/VirtualMachine_Restart_MaximumSet_Gen.js
const { ComputeManagementClient } = require("@azure/arm-compute");
const { DefaultAzureCredential } = require("@azure/identity");

/**
 * This sample demonstrates how to the operation to restart a virtual machine.
 *
 * @summary the operation to restart a virtual machine.
 * x-ms-original-file: 2025-11-01/virtualMachineExamples/VirtualMachine_Restart_MaximumSet_Gen.json
 */
async function virtualMachineRestartMaximumSetGen() {
  const credential = new DefaultAzureCredential();
  const subscriptionId = "{subscription-id}";
  const client = new ComputeManagementClient(credential, subscriptionId);
  await client.virtualMachines.restart("rgcompute", "aaaaaaaaaaaaaaaaaaaa");
}
